src/model/generateRestHandlers.ts

```diff
@@ -185,7 +186,7 @@ export function generateRestHandlers<
           const id = req.params[primaryKey]
           const where: WeakQuerySelectorWhere<typeof primaryKey> = {
             [primaryKey]: {
-              equals: id,
+              equals: typeof primaryKeyValue === 'number' ? +id : id as any,
```

Comment:
We seem to be repeating this logic quite a lot. Basically, whenever we're retrieving the actual primary key value from the `req.params`, we need to coerce it to number if that's its type in the model definition. 

Perhaps, let's abstract this coercion into a function?

```js
const primaryKeyValue = modelDefinition[primaryKey].getValue()

function extractPrimaryKey(params: Record<string, string>): PrimaryKeyType {
  const raw = params[primaryKey]
  return isNaN(primaryKeyValue) ? raw : Number(raw)
}

// ...
const id = extractPrimaryKey(req.params)
{
  [primaryKey]: { equals: id }
}
```

Reply:
That sounds good!

Reply:
Changes pushed.